core/iwasm/common/wasm_memory.c

```diff
@@ -685,19 +699,39 @@ wasm_enlarge_memory_internal(WASMModuleInstance *module, uint32 inc_page_count)
     os_writegsbase(memory_data_new);
 #endif
 
+return_func:
+    if (!ret && enlarge_memory_error_cb) {
+#if WASM_ENABLE_INTERP != 0
+        enlarge_memory_error_cb(
+            inc_page_count, total_size_old, 0, failure_reason,
+            (WASMModuleInstanceCommon *)module,
+            ((WASMModuleInstanceExtra *)module->e)->common.cur_exec_env);
+#endif
+#if WASM_ENABLE_AOT != 0
+        enlarge_memory_error_cb(
+            inc_page_count, total_size_old, 0, failure_reason,
+            (WASMModuleInstanceCommon *)module,
+            ((AOTModuleInstanceExtra *)module->e)->common.cur_exec_env);
+#endif
```

Comment:
The module instance can be AOTModuleInstance when macro WASM_ENABLE_INTERP is 1, suppose WASM_ENABLE_AOT is also enabled. Had better change to be like:
```C
    if (!ret && enlarge_memory_error_cb) {
        WASMExecEnv *exec_env = NULL;

#if WASM_ENABLE_INTERP != 0
        if (module->module_type == Wasm_Module_Bytecode)
            exec_env = ((WASMModuleInstanceExtra *)module->e)->common.cur_exec_env;
#endif
#if WASM_ENABLE_AOT != 0
        if (module->module_type == Wasm_Module_AoT)
            exec_env = ((AOTModuleInstanceExtra *)module->e)->common.cur_exec_env;
#endif

        enlarge_memory_error_cb(
            inc_page_count, total_size_old, 0, failure_reason,
            (WASMModuleInstanceCommon *)module, exec_env);
    }
```